### ISSUE TITLE
Portuguese from Portugal translation

### ISSUE THREAD

**Issue by fabio-silva-pt** [BuIlDaLiBlE/BetterHI3Launcher#27]
Are you interested in a Portuguese (from Portugal not Brazil) translation?

Portugal doesn't have a lot of people and Portuguese from Brazil is similar, but in case you want I'm here available and with fast responses.

**Comment by BuIlDaLiBlE** [comment]
Sure, why not! Here's the Brazil Portuguese with all relevant text:
<details>
  <summary>Spoiler</summary>
  
```
{
  "pt": {
    "version": "Versão",
    "outdated": "Desatualizado",
    "enabled": "Ativado",
    "disabled": "Desativado",
    "button_yes": "Sim",
    "button_no": "Não",
    "button_download": "Baixar",
    "button_downloading": "Baixando",
    "button_running": "Rodando",
    "button_update": "Atualizar",
    "button_pause": "Pausar",
    "button_launch": "Iniciar",
    "button_options": "Opções",
    "button_resume": "Retomar",
    "button_confirm": "Confirmar",
    "button_cancel": "Cancelar",
    "button_github": "Ir para o repositório GitHub",
    "button_generate": "Gerar",
    "label_server": "Servidor",
    "label_mirror": "Espelho",
    "label_log": "Mostrar log",
    "label_preload": "Pré-download do jogo",
    "label_get_now": "Obter já",
    "label_verifying": "Examinando",
    "label_paused": "Em pausa",
    "label_done": "Concluído",
    "label_retry": "Tentar de novo",
    "label_downloaded_1": "Baixado {0}%",
    "label_downloaded_2": "Baixando recursos:",
    "label_eta": "Tempo estimado:",
    "label_speed": "Velocidade de baixada:",
    "introbox_title": "Bem-vindo ao Better HI3 Launcher!",
    "introbox_msg": "Parece que essa é a sua primeira vez usando esse laucher. Estou feliz que você tenha decidido testar ele, então nesse caso, caso queira dar umas sugestões, você pode clicar em \"Opções\" e depois em \"Enviar feedback\".\nSe você já tem o jogo instalado, apenas clique em \"Download\" e mostre onde seu jogo está. O launcher irá detectar a versão do jogo automaticamente e você não precisará baixá-lo novamente.",
    "downloadcachebox_msg": "Selecione se deseja baixar o pacote de cache completo ou apenas arquivos numéricos.\nSelecione\"Cache completo\" se você tiver problemas para atualizar os recursos do evento.\nSelecione \"Arquivos numéricos\" se você tiver problemas para atualizar as configurações.\nPor favor, note que atualmente não há como recuperar automaticamente o cache mais recente e temos que carregá-lo manualmente para um espelho.\nUsando espelho: {0}.\nCache atualizado por último: {1}.\nO mantenedor do espelho atual é {2}.",
    "downloadcachebox_button_full_cache": "Cache completo",
    "downloadcachebox_button_numeric_files": "Arquivos numéricos",
    "repairbox_msg": "Isso verificará todos os arquivos do jogo e tentará reparar se algum estiver quebrado. Pode demorar um pouco.\nContinuar?\nEspelhos disponíveis: {0}.\nO mantenedor do espelho atual é {1}.",
    "fpsinputbox_title": "Insira o limite de FPS customizado",
    "fpsinputbox_label_combatfps": "FPS em jogo",
    "fpsinputbox_label_menufps": "FPS no menu",
    "resolutioninputbox_title": "Insira a resolução customizada",
    "resolutioninputbox_label_width": "Largura",
    "resolutioninputbox_label_height": "Altura",
    "resolutioninputbox_label_fullscreen": "Tela cheia",
    "changelogbox_1_msg": "Better HI3 Launcher acaba de ficar ainda melhor. Aqui está o que mudou:",
    "changelogbox_2_msg": "Carregando o changelog...",
    "changelogbox_3_msg": "Não foi possível carregar o changelog.",
    "aboutbox_msg": "Bem, é muito mais avançado, não é? :^)\nEste projeto foi feito com a esperança de que muitos capitães tivessem uma melhor experiência com o jogo.\nEle não é afiliado à miHoYo e é totalmente código-aberto.\nQualquer comentário será muito bem vindo.\nAgradecimentos especiais à esses contribuidores do GitHub:\nSinsOfSeven - Contribuição de resolução personalizada\nProxy-E23 - Tradução da idioma espanhola\nSpookyKisuy - Tradução da idioma portuguêsa\nbulawin1, Vrontis - Tradução da idioma alemã\nKorewaLidesu - Tradução da idioma vietnamita\nSm1leAway - Tradução da idioma sérvia\nVoRTeXStarz - Tradução da idioma tailandesa",
    "contextmenu_download_cache": "Baixar cache",
    "contextmenu_fix_subtitles": "Corrigir legendas",
    "contextmenu_repair": "Reparar arquivos do jogo",
    "contextmenu_move": "Mover arquivos do jogo",
    "contextmenu_uninstall": "Desinstalar jogo",
    "contextmenu_game_settings": "Gerenciar configurações de jogo",
    "contextmenu_custom_fps": "Definir limite de FPS personalizado",
    "contextmenu_custom_resolution": "Definir resolução personalizada",
    "contextmenu_reset_download_type": "Redefinir tipo de download",
    "contextmenu_reset_game_settings": "Redefinir as configurações do jogo",
    "contextmenu_web_profile": "Entrar no website",
    "contextmenu_feedback": "Enviar feedback",
    "contextmenu_changelog": "Mostrar changelog",
    "contextmenu_custom_background": "Alterar plano de fundo",
    "contextmenu_sounds": "Sons",
    "contextmenu_language": "Idioma",
    "contextmenu_language_system": "Padrão do sistema",
    "contextmenu_language_english": "Inglês",
    "contextmenu_language_russian": "Russo",
    "contextmenu_language_spanish": "Espanhol",
    "contextmenu_language_portuguese": "Português",
    "contextmenu_language_german": "Alemã",
    "contextmenu_language_vietnamese": "Vietnamita",
    "contextmenu_language_serbian": "Sérvia",
    "contextmenu_language_thai": "Tailandês",
    "contextmenu_language_contribute": "Quer ajudar?",
    "contextmenu_about": "Sobre",
    "progresstext_checking_update": "Verificando atualizações...",
    "progresstext_cleaning_up": "Limpando...",
    "progresstext_downloaded": "Baixado {0}/{1} ({2})",
    "progresstext_downloading_file": "Baixando arquivo {0}/{1}...",
    "progresstext_download_size": "Tamanho do download",
    "progresstext_error": "Erros foram encontrados :^(",
    "progresstext_eta": "Tempo estimado: {0}",
    "progresstext_fetching_hashes": "Buscando hashes de arquivo...",
    "progresstext_generating_hash": "Gerando hash para arquivo {0}/{1}...",
    "progresstext_initiating_download": "Iniciando download...",
    "progresstext_mirror_connect": "Conectando ao espelho...",
    "progresstext_moving_files": "Movendo arquivos do jogo...",
    "progresstext_uninstalling": "Desinstalando o jogo...",
    "progresstext_unpacking_1": "Extraindo arquivos do jogo...",
    "progresstext_unpacking_2": "Extraindo arquivo do jogo {0}/{1}...",
    "progresstext_updating_launcher": "Atualizando launcher...",
    "progresstext_verifying_file": "Verificando arquivo do jogo {0}/{1}...",
    "progresstext_verifying_files": "Verificando arquivos do jogo...",
    "progresstext_zipping": "Adicionando arquivo {0}/{1} para arquivo ZIP...",
    "msgbox_abort_1_msg": "Tem certeza de que deseja cancelar o download e fechar o launcher?",
    "msgbox_abort_2_msg": "O progresso não será salvo.",
    "msgbox_abort_3_msg": "O progresso será salvo.",
    "msgbox_abort_title": "Pedido de aborto",
    "msgbox_conn_bp_error_msg": "Não foi possível conectar-se ao Bp Network:",
    "msgbox_conn_mihoyo_error_msg": "Não foi possível conectar-se aos servidores miHoYo:",
    "msgbox_custom_background_1_msg": "Você pode selecionar um arquivo para colocar como plano de fundo.\nPode ser uma imagem ou um vídeo.\nResolução recomendada é de {0} por {1} pixels.\nContinuar?",
    "msgbox_custom_background_2_msg": "Selecione o que você gostaria de fazer com o plano de fundo.",
    "msgbox_custom_background_3_msg": "O arquivo selecionado não tem a resolução ideal. Você ainda pode usá-lo, mas ele será esticado para o tamanho da janela.\nContinuar?",
    "msgbox_custom_background_4_msg": "Por favor selecione outro arquivo.",
    "msgbox_custom_background_5_msg": "O arquivo é muito grande.\nVocê pode selecionar arquivos de no máximo {0}.",
    "msgbox_custom_background_6_msg": "Não há espaço suficiente na unidade para usar o arquivo.",
    "msgbox_custom_background_7_msg": "O arquivo selecionado não tem um formato compatível.",
    "msgbox_custom_background_8_msg": "Falha ao carregar o arquivo selecionado.",
    "msgbox_custom_background_edit": "Alterar",
    "msgbox_custom_background_delete": "Redefinir para o oficial",
    "msgbox_custom_fps_1_msg": "Os valores não devem estar vazios.",
    "msgbox_custom_fps_2_msg": "Os valores não devem serem menores que zero.",
    "msgbox_custom_fps_3_msg": "Valores inferiores a 30 não são recomendados. Continuar?",
    "msgbox_custom_fps_4_msg": "Limite de FPS em jogo e menu definidos para {0} e {1}, respectivamente.",
    "msgbox_custom_resolution_1_msg": "Altura maior que largura não é recomendado.\nContinuar?",
    "msgbox_custom_resolution_2_msg": "Resolução definida para {0}x{1} com tela cheia {2}.",
    "msgbox_download_cache_1_msg": "O cache completo está prestes a ser baixado.",
    "msgbox_download_cache_2_msg": "Os arquivos numéricos estão prestes a serem baixados.",
    "msgbox_download_cache_3_msg": "Tamanho do download: {0}.\nContinuar?",
    "msgbox_download_cache_4_msg": "O cache está desatualizado parcial ou totalmente.\nVerifique a hora da última atualização para a data exata.\nVocê ainda quer atualizar?",
    "msgbox_download_type_1_msg": "Isso irá redefinir o tipo de download de recursos do jogo em uma tentativa de consertar o infame loop de atualização que não permite que você entre no jogo.\nContinuar?",
    "msgbox_download_type_2_msg": "O tipo de download foi redefinido. Inicie o jogo e selecione \"HD assets\".",
    "msgbox_download_type_3_msg": "O tipo de download já foi redefinido. Inicie o jogo e selecione \"HD assets\".",
    "msgbox_extract_skip_msg": "A extração terminou, mas alguns arquivos não foram extraídos. Você pode querer extraí-los manualmente.\nPara mais informações dê uma olhada no log.",
    "msgbox_extract_skip_title": "Aviso de salto de arquivo",
    "msgbox_fix_subtitles_1_msg": "Isso tentará corrigir as legendas dos CGs (e banners de gacha). Certifique-se de que já baixou todos os CGs do jogo.\nContinuar?",
    "msgbox_fix_subtitles_2_msg": "Extraindo arquivo de legenda {0}/{1}...",
    "msgbox_fix_subtitles_3_msg": "Verificando arquivo de legenda {0}/{1}...",
    "msgbox_fix_subtitles_4_msg": "Legendas extraídas para {0} CGs.",
    "msgbox_fix_subtitles_5_msg": "Corrigido {0} arquivos de legenda",
    "msgbox_fix_subtitles_6_msg": "Nenhum arquivo de legenda foi corrigido. Eles ainda não foram baixados ou já estão corrigidos.",
    "msgbox_game_download_error_msg": "Ocorreu um erro ao baixar os arquivos do jogo.\nPara mais informações dê uma olhada no log.",
    "msgbox_game_download_error_title": "Erro ao baixar arquivos de jogo",
    "msgbox_game_download_mirror_error_msg": "Ocorreu um erro ao baixar do espelho.\nPara mais informações dê uma olhada no log.",
    "msgbox_game_download_mirror_old_msg": "Parece que a versão do jogo nos servidores miHoYo é mais recente do que a do espelho.\nNão há motivo para baixar uma versão desatualizada, peça ao mantenedor do espelho para fazer o upload de uma nova versão.",
    "msgbox_game_download_paused_msg": "O jogo ainda não foi totalmente baixado. Mudar de espelho ou servidor irá reiniciar o progresso do download.\nContinuar?",
    "msgbox_generic_error_msg": "Ocorreu um erro.\nPara obter informações dê uma olhada no log.",
    "msgbox_generic_error_title": "Erro",
    "msgbox_install_dir_error_msg": "Ocorreu um erro ao selecionar o diretório de instalação do jogo.\nPara mais informações dê uma olhada no log.",
    "msgbox_install_dir_error_title": "Diretório inválido",
    "msgbox_install_error_msg": "Ocorreu um erro ao instalar os arquivos do jogo.\nPara mais informações dê uma olhada no log.",
    "msgbox_install_error_title": "Erro de instalação",
    "msgbox_install_existing_dir_invalid_msg": "O diretório selecionado não contém uma instalação válida do jogo. Este launcher só oferece suporte a clientes globais e SEA.",
    "msgbox_install_existing_dir_msg": "O jogo parece já ter sido instalado em:\n{0}\nUsar este diretório?",
    "msgbox_install_existing_no_local_version_msg": "Não foi possível determinar a versão local.\nSeu jogo já está atualizado? Por favor, escolha com sabedoria!\nSelecionar \"Sim\" fará com que você seja capaz de iniciar o jogo.\nSelecionar \"Não\" fará com que você tenha que baixar o jogo.",
    "msgbox_install_little_space_msg": "Potencialmente, não há espaço livre suficiente no dispositivo selecionado, é recomendável liberar algum espaço ou a instalação pode resultar em falha.\nContinuar?",
    "msgbox_install_msg": "O jogo será instalado em:\n{0}\nContinuar?",
    "msgbox_install_title": "Aviso de instalação",
    "msgbox_install_wrong_drive_type_msg": "Não é possível instalar no dispositivo selecionado.",
    "msgbox_language_msg": "O idioma será alterado para {0} e o launcher será reiniciado.\nContinuar?",
    "msgbox_launcher_download_error_msg": "Ocorreu um erro ao baixar o launcher.\nPara mais informações dê uma olhada no log.",
    "msgbox_mirror_error_msg": "Ocorreu um erro com o espelho. Peça ao mantenedor do espelho para chegar ao final disso.\nMensagem: {0}",
    "msgbox_mirror_info_msg": "Use este espelho apenas se você não puder baixar o jogo através dos servidores miHoYo.\nNote que ele é atualizado manualmente.\nContinuar?",
    "msgbox_move_1_msg": "O jogo será movido para:\n{0}\nContinuar?",
    "msgbox_move_2_msg": "Não é possível mover o jogo dentro do diretório do jogo atual.",
    "msgbox_move_3_msg": "Não é possível mover o jogo enquanto o launcher está dentro do diretório do jogo. Mova o launcher para fora do diretório e tente novamente.",
    "msgbox_move_error_title": "Mover erro",
    "msgbox_move_little_space_msg": "Potencialmente, não há espaço livre suficiente no dispositivo selecionado, é recomendável liberar algum espaço ou a operação de movimentação pode resultar em falha.\nContinuar?",
    "msgbox_move_title": "Mover",
    "msgbox_move_wrong_drive_type_msg": "Não é possível mover para o dispositivo selecionado.",
    "msgbox_net_error_msg": "Ocorreu um erro ao conectar ao servidor:\n{0}",
    "msgbox_net_error_title": "Erro de rede",
    "msgbox_net_version_old_msg": "Este launcher requer que o .NET Framework 4.6.1 ou mais recente esteja instalado.",
    "msgbox_no_game_dir_msg": "O diretório do jogo não foi encontrado.\nTente reinstalar o jogo.",
    "msgbox_no_game_dir_title": "Nenhum jogo diretório",
    "msgbox_no_game_exe_msg": "O executável do jogo não foi encontrado.\nTente reinstalar o jogo.",
    "msgbox_no_game_exe_title": "Nenhum jogo executável",
    "msgbox_no_video_dir_msg": "A pasta de vídeo não foi encontrada.\nTente reinstalar o jogo.",
    "msgbox_notice_title": "Aviso",
    "msgbox_pre_install_msg": "É possível entrar no jogo durante o pré-download dos recursos. Deseja mesmo fazer o pré-download?",
    "msgbox_process_start_error_msg": "Ocorreu um erro ao iniciar o processo.\nPara mais informações dê uma olhada no log.",
    "msgbox_registry_empty_1_msg": "Nenhum valor a ser ajustado no registro existe.",
    "msgbox_registry_empty_2_msg": "Você já executou o jogo?",
    "msgbox_registry_empty_3_msg": "Tente alterar as configurações de vídeo no jogo antes (desativando tela cheia, alterando a predefinição de vídeo, etc.).",
    "msgbox_registry_error_msg": "Ocorreu um erro ao acessar o registro.\nPara mais informações dê uma olhada no log.",
    "msgbox_registry_error_title": "Erro de registro",
    "msgbox_repair_1_msg": "A versão do jogo instalado difere das informações de reparo no servidor. Por termos que atualizar essas informações manualmente, elas podem ficar indisponíveis por algum tempo.\nPor favor, tente novamente mais tarde.",
    "msgbox_repair_2_msg": "Nenhum arquivo precisa de reparos.",
    "msgbox_repair_3_msg": "Encontrados {0} arquivo(s) corrompido(s)/perdido(s). Eles precisam ser reparados baixando-os novamente.\nComo a tecnologia é incrível, apenas os arquivos necessários serão baixados.\nTamanho do download: {1}.\nContinuar?",
    "msgbox_repair_4_msg": "{0} arquivo(s) reparado(s) com sucesso.",
    "msgbox_repair_5_msg": "{0} arquivo(s) não foram reparados. Considere verificar seu dispositivo de armazenamento, isso pode ser o problema.",
    "msgbox_repair_6_msg": "Isso irá gerar um arquivo JSON contendo a lista de arquivos do jogo e seus hashes, então criar um arquivo ZIP com esses arquivos.\nContinuar?",
    "msgbox_start_error_msg": "Ocorreu um erro ao iniciar o launcher:\n{0}",
    "msgbox_start_error_title": "Erro de inicialização",
    "msgbox_reset_game_settings_moved_msg": "Esta opção foi movida para desinstalação.\nVocê gostaria de ir lá agora?",
    "msgbox_unhandled_exception_msg": "Ops, algo muito ruim aconteceu e o launcher deve ser fechado. Mas você pode ajudar!\nHá um arquivo de log que você pode enviar para ajudar a diagnosticar o erro.\nVocê quer abrir este arquivo?",
    "msgbox_uninstall_1_msg": "Selecione o que você gostaria de desinstalar.",
    "msgbox_uninstall_2_msg": "Tem certeza de que deseja desinstalar isso?",
    "msgbox_uninstall_3_msg": "Tem certeza de que deseja desinstalar isso? :^(",
    "msgbox_uninstall_4_msg": "Esta ação é irreversível.\nApenas faça isso se estiver tendo problemas com o jogo!\nVocê ainda quer continuar?",
    "msgbox_uninstall_5_msg": "Não é possível desinstalar o jogo enquanto o launcher está dentro do diretório do jogo. Mova o launcher para fora do diretório e tente novamente.",
    "msgbox_uninstall_6_msg": "Desinstalado com sucesso:",
    "msgbox_uninstall_error_msg": "Ocorreu um erro ao desinstalar o jogo.\nPara mais informações dê uma olhada no log.",
    "msgbox_uninstall_error_title": "Erro de desinstalação",
    "msgbox_uninstall_game_files": "Arquivos do jogo",
    "msgbox_uninstall_game_cache": "Cache do jogo",
    "msgbox_uninstall_game_settings": "Configurações do jogo",
    "msgbox_uninstall_title": "Desinstalar",
    "msgbox_update_check_error_msg": "Ocorreu um erro ao verificar atualizações.\nPara mais informações dê uma olhada no log.",
      "msgbox_update_check_error_title": "Erro de verificação de atualização",
      "msgbox_update_title": "Aviso de atualização",
      "msgbox_verify_error_1_msg": "Ocorreu um erro durante o download. Por favor, tente novamente.",
      "msgbox_verify_error_2_msg": "Ocorreu um erro durante o download. O arquivo pode estar corrompido.\nContinuar do mesmo jeito?",
      "msgbox_verify_error_title": "Erro de validação de arquivo"
    }
  }
  ```
  
</details>

**Comment by BuIlDaLiBlE** [comment]
Implemented in [v1.2.20210622.0](https://github.com/BuIlDaLiBlE/BetterHI3Launcher/releases/tag/1.2.20210622.0).
